### ISSUE TITLE
`make generate` will fail if directory name has space in it

### ISSUE THREAD

**Issue by lauchokyip** [kubernetes-sigs/kubebuilder#3434]
### What broke? What's expected?
Broke:
make generate will fail if directory name has space in it

Expected:
`make generate`  runs successfully regardless if there's space for the directory or not

<details>

```
Writing scaffold for you to edit...
controllers/laker_controller.go
Update dependencies:
$ go mod tidy
make[1]: Entering directory '/home/lau/foo bar/kubebuilder/testdata/project-v2'
Makefile:120: warning: overriding recipe for target '/home/lau/foo'
Makefile:106: warning: ignoring old recipe for target '/home/lau/foo'
Makefile:125: warning: overriding recipe for target '/home/lau/foo'
Makefile:120: warning: ignoring old recipe for target '/home/lau/foo'
make[1]: Circular /home/lau/foo <- /home/lau/foo dependency dropped.
mkdir -p /home/lau/foo bar/kubebuilder/testdata/project-v2/bin
make[1]: Circular /home/lau/foo <- /home/lau/foo dependency dropped.
test -s /home/lau/foo bar/kubebuilder/testdata/project-v2/bin/controller-gen || GOBIN=/home/lau/foo bar/kubebuilder/testdata/project-v2/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0
/bin/sh: 1: test: /home/lau/foo: unexpected operator
/bin/sh: 1: bar/kubebuilder/testdata/project-v2/bin: Permission denied
make[1]: *** [Makefile:125: /home/lau/foo] Error 126
make[1]: Leaving directory '/home/lau/foo bar/kubebuilder/testdata/project-v2'
make: *** [Makefile:72: generate-testdata] Error 2
```

</details>

### Reproducing this issue

```
> mkdir foo\ bar
> cd foo\ bar
> git clone https://github.com/kubernetes-sigs/kubebuilder.git
> cd kubebuilder
> make generate
```

### KubeBuilder (CLI) Version

N/A

**Comment by camilamacedo86** [comment]
Are you speaking about the make generate target used for us to develop and keep the project maintained?
If so, it is fine because we ought to not have a directory with spaces in the name

**Comment by stormqueen1990** [comment]
Hello there, @camilamacedo86! 👋

I'm able to reproduce this issue using both `make generate` and `kubebuilder create api` as long as there is at least one directory in the tree with a whitespace in its name. I tested with a path similar to `/projects/a kubebuilder test/kubebuilder-test`, since `kubebuilder init` requires the current folder to not have whitespace in the name.

I noticed I can partly address this by quoting the `LOCALBIN` variable assignment in the Makefile generators, but it's just a partial fix because Makefile does not deal well with targets containing whitespace and some of the targets in the Makefile are named with the path of the binaries. I'm not an expert in Makefile, so I'm unsure if renaming those targets might break something, but I can create a PR with my partial fix if the project sees value in it.
